### PR TITLE
Configuration for google cloud build to run tests.

### PR DESCRIPTION
**Pull request by bgogul**
All the necessary configuration is under `gcb/` directory. I went back and forth between kokoro and gcb, but decided to stick with gcb in the end for a variety of reasons:
  - The setup is working now (modulo providing view permissions for others contributors).
  - Integrating ResultsUI and RBE on Kokoro requires further setup and seems more involved.
  - With GCB all the necessary configuration is in the open source code itself. 
 
One of the advantages of Kokoro is that it posts nice status messages with an embedded link to the test ResultsUI. However, I figured out a way to do the same in GCB. It just requires wrapping the bazel runs in a script, extracting the sponge links from the output and posting as status. I will do that in a follow up URL.

### CODE REVIEW

File: gcb/rbe/configs/path/cc/tools/cpp/empty.cc [hunk 1]
```diff
@@ -0,0 +1 @@
+int main() {}
```
Comment (markww) [inline gcb/rbe/configs/path/cc/tools/cpp/empty.cc:1]: No newline at the end of file.

File: gcb/rbe/configs/path/cc/cc_wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Ship the environment to the C++ action
+#
+set -eu
+
+# Set-up the environment
+
+
+# Call the C++ compiler
+/usr/bin/clang "$@"
```
Comment (markww) [inline gcb/rbe/configs/path/cc/cc_wrapper.sh:25]: Is this a sample file? Should there be an environment setup?

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.vscode
+.git
+bazel-out
+bazel-arcs
+bazel-bin
+
```
Comment (markww) [inline .dockerignore:6]: Extraneous blank line. Ignore if this is just a copy of a sample file.

File: gcb/rbe/configs/path/cc/BUILD [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    "-fstack-protector",
+    "-Wall",
+    "-Wthread-safety",
+    "-Wself-assign",
+    "-fcolor-diagnostics",
+    "-fno-omit-frame-pointer"],
+    opt_compile_flags = ["-g0",
+    "-O2",
+    "-D_FORTIFY_SOURCE=1",
+    "-DNDEBUG",
```
Comment (markww) [inline gcb/rbe/configs/path/cc/BUILD:113]: Should we be turning assertions off? Usually I like to have them on at runtime, but if this is a file we got from elsewhere, this is fine.

**Review by markww — COMMENTED** [review]
Found some basic style stuff, but nothing serious. Then again, this is a big pile of stuff and I'm not sure what parts I should look at in depth and what I should skim. It looks to me like much of this are copies of standard files; is there a particular directory pattern I should skim rather than look at in depth?

**Comment by bgogul** [comment]
> Found some basic style stuff, but nothing serious. Then again, this is a big pile of stuff and I'm not sure what parts I should look at in depth and what I should skim. It looks to me like much of this are copies of standard files; is there a particular directory pattern I should skim rather than look at in depth?

Everything under the `gcb/rbe` is generated. Feel free to skip them. I will probably avoid incorporating any of the style changes there. It might be worthwhile looking at the other files in some detail. Hopefully, there are no surprises there.

File: gcb/presubmit.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Run all the tests.
+${BAZEL_TEST} //src/...
+# Verifies that the following targets build fine:
+#  - Arcs parser and proto works.
+#  - Arcs manifest tests that we do not yet handle parse correctly.
+#  - Authorization logic code with bazel.
+#  - Arcs manifest examples.
+${BAZEL_BUILD} //third_party/arcs/examples:consume third_party/arcs/proto:manifest_cc_proto \
+   //src/analysis/souffle/tests/arcs_manifest_tests_todo/... \
+   //rust/tools/authorization-logic:authorization_logic \
```
Comment (markww) [inline gcb/presubmit.sh:21]: Now that we are actually using the authorization logic in tests, this should probably be removed.

Actually, should we swap the order of the build and test invocations? In a sense, the build checks are "more basic" in that they just make sure that everything still builds without invoking any deep Raksha functionality. Therefore, we would expect these build commands to pass more frequently than the test commands, making them more cacheable.

**Review by markww — APPROVED** [review]

File: gcb/rbe/configs/path/cc/tools/cpp/empty.cc (hunk 1 above)
Comment (bgogul) [reply gcb/rbe/configs/path/cc/tools/cpp/empty.cc:1]: Auto generated. Will leave this as is.

File: .dockerignore (hunk 3 above)
Comment (bgogul) [reply .dockerignore]: Done.

File: gcb/rbe/configs/path/cc/BUILD (hunk 4 above)
Comment (bgogul) [reply gcb/rbe/configs/path/cc/BUILD:113]: Auto generated. I will leave this as is for now.

File: gcb/rbe/configs/path/cc/cc_wrapper.sh (hunk 2 above)
Comment (bgogul) [reply gcb/rbe/configs/path/cc/cc_wrapper.sh:25]: Auto generated.

File: gcb/presubmit.sh (hunk 5 above)
Comment (bgogul) [reply gcb/presubmit.sh]: done.

**Pull request merged**